pkg/controller/worker/helper.go

```diff
@@ -58,3 +62,31 @@ func (w *workerDelegate) updateWorkerProviderStatus(ctx context.Context, workerS
 		return nil
 	})
 }
+
+func (w *workerDelegate) updateMachineDependenciesStatus(ctx context.Context, workerStatus *api.WorkerStatus, ServerGroupDependencies []api.ServerGroupDependency, err error) error {
+	workerStatus.ServerGroupDependencies = ServerGroupDependencies
+	if statusUpdateErr := w.updateWorkerProviderStatus(ctx, workerStatus); statusUpdateErr != nil {
+		if err != nil {
+			err = errors.Wrapf(statusUpdateErr, err.Error())
+		} else {
+			err = statusUpdateErr
+		}
+	}
+
+	return err
+}
+
+// ClusterTechnicalName returns the technical name of the cluster this worker belongs.
+func (w *workerDelegate) ClusterTechnicalName() string {
+	return w.cluster.ObjectMeta.Name
+}
+
+func (w *workerDelegate) findMatchingServerGroupPolicy(policy string, allowedPolicies []string) bool {
```

Comment:
Does it make sense to move this into the API helper package? (actually, is this even called somewhere? I cannot find it 🔍)

Reply:
Nice catch. That's an artifact from before I add the validation for the worker specs